surfinBH/_fit_evaluators/fit_7dq4Emri.py

```diff
@@ -0,0 +1,267 @@
+                np.random.choice([0, np.pi])
+        else:
+            chiAth = np.arccos(np.random.uniform(-1., 1.))
+            chiBth = np.arccos(np.random.uniform(-1., 1.))
+            chiAph = np.random.uniform(0, 2*np.pi)
+            chiBph = np.random.uniform(0, 2*np.pi)
+
+        chiA = [chiAmag*np.sin(chiAth)*np.cos(chiAph),
+                chiAmag*np.sin(chiAth)*np.sin(chiAph),
+                chiAmag*np.cos(chiAth)]
+
+        chiB = [chiBmag*np.sin(chiBth)*np.cos(chiBph),
+                chiBmag*np.sin(chiBth)*np.sin(chiBph),
+                chiBmag*np.cos(chiBth)]
+
+        return q, chiA, chiB
+
+    #-------------------------------------------------------------------------
+    def _eval_wrapper(self, fit_key, q, chiA, chiB, **kwargs):
+        """ Evaluates the NRSur7dq4EmriRemnant model. """
+        chiA = np.array(chiA)
+        chiB = np.array(chiB)
+
+        # Warn/Exit if extrapolating
+        allow_extrap = kwargs.pop('allow_extrap', False)
+        if allow_extrap: warnings.warn("""Optional argument 'allow_extrap' is 
+        unused for this model. It works at arbitrarily large mass ratios.""")
+        self._check_param_limits(q, chiA, chiB, True)
+
+        x = np.concatenate(([q], chiA, chiB))
+
+        def eval_r_isco(chiz):
+            """
+            Estimate ISCO radius following Eq. (2.21) by Bardeen et al. 
+            (https://doi.org/10.1086/151796)
+            """
+ 
+            Z1 = 1+(1-chiz**2)**(1/3)*((1+chiz)**(1/3)+(1-chiz)**(1/3))
+            Z2 = (3*chiz**2+Z1**2)**(1/2)
+            r_ISCO = 3+Z2-np.sign(chiz)*((3-Z1)*(3+Z1+2*Z2))**(1/2)
+    
+            return r_ISCO
+
+        def eval_emriChif(q, chiA):
+            """
+            Estimate remnant spin for EMRI binary following Eq. (7) 
+            presented in THE PAPER.    
+            """
+            r_ISCO = eval_r_isco(chiA[2])
+            E_ISCO = np.sqrt(1-2/(3*r_ISCO))
+            L_ISCO = np.array([0,0,2/(3*3**(1/2))*(1+2*np.sqrt(3*r_ISCO-2))])
+                      
+            return chiA + 1/q*(L_ISCO-2*chiA*E_ISCO)  
+
+        def extrap_chif(x, fit_key):
+            """
+            The model evaluates remnant spin using GPR fit up to the upper bound
+            of the EMRI training region (q = 1000). At q > 1000, it returns the 
+            expected EMRI limit (Eq. (7) in THE PAPER). There is a transition 
+            function (Eq. (9) in THE PAPER) connecting smoothly these 
+            two regions. The transition takes place from q=1000 to q=2000.
+            """
+            x_i = x.copy()
+            x_i[0] = self.qEmriMax
+            
+            """This if statement defines the width of the transition region"""
+            if x[0] < 2*self.qEmriMax:
+                tmp = self._evaluate_fits(x_i, fit_key)
+                y_i, y_i_err = tmp.T[0], tmp.T[1]
+                y_f = eval_emriChif(2*self.qEmriMax, x[1:4])
+                y_f_err = 2*np.abs(x[1:4] - y_f)
```

Comment:
Where is this coming from? You're subtracting chif from chi1? Why? These are the kinds of things that definitely need a comment :)

Reply:
Done

Reply:
By "finite q", do you mean self.qEmriMax? It's better to just say that.

Reply:
How well thought out is this error estimate? Should people take it seriously? Is it too conservative, and might scare people? I don't remember a discussion about this, so just want to make sure. Also asking @dgerosa.

Another approach could be we return 0 for the error estimate when q>self.qEmriMax because we don't have a very good way to estimate the error. 

Either way, what we are doing for the error estimate in different regimes should be mentioned in the documentation, perhaps after line 81 above?